### PR TITLE
Allow to gather wool staples from sheeps

### PR DESCRIPTION
**Pull request by illi-kun**
...and process them to felt patches.

Very basic processing of the wool staples to the felt patches is implemented for now. It can be enhanced in future (with additional operations, tools and so on) if anyone wants to do so.

Related to #14211.

### CODE REVIEW

File: data/json/recipes/recipe_others.json [hunk 1]
```diff
@@ -146,6 +146,25 @@
...
+  "result": "felt_patch",
+  "category": "CC_OTHER",
+  "subcategory": "CSC_OTHER_OTHER",
+  "skill_used": "fabrication",
+  "skills_required" : [ "tailor", 1 ],
+  "difficulty": 1,
+  "time": 15000,
+  "reversible": false,
+  "autolearn": true,
+  "components": [
```
Comment (chaosvolt) [inline data/json/recipes/recipe_others.json]: Interesting, though at the moment the recipe requires soap and water, but no wool staples. o.O

Comment (illi-kun) [inline data/json/recipes/recipe_others.json]: Doh! >_<'

Good catch, thank you.

Comment (illi-kun) [inline data/json/recipes/recipe_others.json]: Fixed & tested.

**Comment by chaosvolt** [comment]
Ah, nice. X3

**Comment by Rivet-the-Zombie** [comment]
We should probably add the ability to shear sheep without killing them at some point, but this is good for now.

**Comment by chaosvolt** [comment]
Agreed, this is a first step.

Him. Maybe if we had an examine option for non-hostile sheep, like with NPCs and neutered zombies?

**Comment by Coolthulhu** [comment]
Basic shearing would be easy:
* Add an option to pet menu to shear when the player has a suitable implement
* Shearing gives wool and adds an effect to the critter that prevents further shearing until it times out

It wouldn't work too well because effects only time out in the reality bubble, but I intend to implement retroactive effect calculations when I have enough free time (I'm currently quite busy with boring IRL stuff, that's why I'm not contributing). Retroactive effect timeout would be all it would take for it to work tolerably.

Same option could be used to milk cows, with the requirement being changed to requirement of a liquid container.
If we wanted "partial milking", the "harvest timeout" effect would need to have varying intensity.

The next improvement to this would be a long action.
The sheep would have to stay within some range for the entire length of the action, possibly getting a "restrained" effect added on every tick. The action would probably have to fail if the critter wasn't where expected.

**Comment by chaosvolt** [comment]
> It wouldn't work too well because effects only time out in the reality bubble, but I intend to implement retroactive effect calculations when I have enough free time (I'm currently quite busy with boring IRL stuff, that's why I'm not contributing). Retroactive effect timeout would be all it would take for it to work tolerably.

Ah, no worries.

And yeah, seems interesting. This could even be expansed further into the foundation of a saddling and riding system, if the options and the general issue of caring livestock was fleshed out alongside it.

**Comment by Chezzo** [comment]
Ooh, then we can do milk from cows! And eggs from chickens!

**Pull request merged**